src/OrleansRuntime/GrainDirectory/GrainDirectoryPartition.cs

```diff
@@ -364,21 +376,58 @@ internal void RemoveGrain(GrainId grain)
         internal AddressesAndTag LookUpActivations(GrainId grain)
```

Comment:
Thought about that optimization, but wanted to process iteratively.

 As for majority - in commit https://github.com/dotnet/orleans/commit/ed1b4efeb31b6fb57159257f06272752da620b13#diff-10c9ac6653bd494c2edbdb7701be8c90 
method <code>LookUpGrain</code> was renamed into <LookUpActivations>, so that it now may seem to be responsible for looking up of multiple activations, which is usually not the case. 

![chrome_2016-09-29_19-16-36](https://cloud.githubusercontent.com/assets/5787619/18962414/6b6e67ae-8679-11e6-9c28-0e66141a4c33.png)